tests/integration/test_api.py

```diff
@@ -383,6 +389,39 @@ def test_annotation_event(client, monkeypatch, httpserver):
     assert result.status_code == 200
 
 
-def test_display_insight_prediction_for_products(client, mocker):
+def test_insight_prediction(client, mocker):
     result = client.simulate_get("/api/v1/insights/predictions")
     assert result.status_code == 200
+    assert result.json == {"count": 0, "predictions": [], "status": "no_predictions"}
+
+    PredictionFactory(barcode="123")
+    result = client.simulate_get("/api/v1/insights/predictions")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 1
+    assert data["status"] == "found"
+    predictions_data = data["predictions"]
+    assert predictions_data[0]["barcode"] == "123"
+    assert predictions_data[0]["type"] == "category"
+    assert predictions_data[0]["value_tag"] == "en:seeds"
+
+    PredictionFactory(value_tag="en:seeds")
+    result = client.simulate_get("/api/v1/insights/predictions")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 2
+    assert data["status"] == "found"
+    predictions_data = data["predictions"]
+    assert predictions_data[0]["type"] == "category"
+    assert predictions_data[1]["type"] == "category"
+    assert predictions_data[0]["value_tag"] == "en:seeds"
+    assert predictions_data[1]["value_tag"] == "en:seeds"
```

Comment:
There were other parameters like `timestamps`, `id`, `data` etc. But I did not use them because they were changing every time the factory was called. And a few other parameters were either `{}` or `None`

Reply:
yes, it's ok not to test every fields in a test. You just have to test, what proves you your test is right.

The id parameter might be important to test. To get it, your must do on line 397 and 408:
```
prediction1 = PredictionFactory(value_tag="en:seeds")
....
prediction2 = PredictionFactory(value_tag="en:seeds")
```
this gives you the created object, so that later on you can do:

```
assert prediction_data[0]["prediction.id"] = prediction1.id
...
assert prediction_data[1]["prediction.id"] = prediction2.id
```